nixops/resources/ebs_volume.py

```diff
@@ -74,12 +76,48 @@ def show_type(self):
     def resource_id(self):
         return self.volume_id
 
+    def get_physical_spec(self):
+        physical = {}
+        if self.volume_id:
+            physical['volumeId'] = self.volume_id
+        return physical
+
+    def prefix_definition(self, attr):
+        return {('resources', 'ebsVolumes'): attr}
 
     def connect(self, region):
         if self._conn: return self._conn
         self._conn = nixops.ec2_utils.connect(region, self.access_key_id)
         return self._conn
 
+    def connect_boto3(self, region):
+        if self._conn_boto3: return self._conn_boto3
+        self._conn_boto3 = nixops.ec2_utils.connect_ec2_boto3(region, self.access_key_id)
+        return self._conn_boto3
+
+    def _check(self, defn):
+        self.connect_boto3(defn.region)
+        try:
+            _vol = self._conn_boto3.describe_volumes(VolumeIds=[defn.volume_id])['Volumes'][0]
```

Comment:
also what if there is no volume_id in the definition like the old behavior ?

Reply:
the function should not run when there is no defined volume_id. the function should not run during a `nixops check` so i assume i should change the name if `_check` if  reserved ?